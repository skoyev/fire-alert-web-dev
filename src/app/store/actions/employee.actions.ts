import { Action } from '@ngrx/store';
import { Employee } from '@appModels/employee';

export enum EmployeeActionTypes {
    ShowNewEmployee = '[ShowNewEmployee] ShowNewEmployee',
    CreateEmployee = '[EmployeeAction] Create Employee',
    CreateEmployeeSuccess = '[CreateEmployeeSuccess] CreateEmployee Success',
    EmployeeError = '[EmployeeError] Error'
}

export class CreateEmployeeSuccess implements Action {
  readonly type = EmployeeActionTypes.CreateEmployeeSuccess;
  constructor(public payload: boolean) {}
}

export class ShowNewEmployeeModal implements Action {
  readonly type = EmployeeActionTypes.ShowNewEmployee;
  constructor(public payload: boolean) {}
}

export class CreateEmployee implements Action {
  readonly type = EmployeeActionTypes.CreateEmployee;
  constructor(public payload: Employee) {}
}

export class EmployeeError implements Action {
  readonly type = EmployeeActionTypes.EmployeeError;
  constructor(public payload: any) {}
}

export type EmployeeActions = 
  CreateEmployee | 
  ShowNewEmployeeModal |
  CreateEmployeeSuccess |
  EmployeeError;
