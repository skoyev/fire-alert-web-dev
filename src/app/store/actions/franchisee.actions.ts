import { Action } from '@ngrx/store';
import { Franchaisee } from '@appModels/franchaisee';

export enum FranchaiseeActionTypes {
    ShowNewFranchaisee = '[ShowNewFranchaisee] New Franchisee',
    CreateFranchaisee = '[FranchaiseeAction] Create Franchaisee',
    CreateFranchaiseeSuccess = '[CreateFranchaiseeSuccess] CreateFranchaisee Success',
    FranchaiseeError = '[FranchaiseeError] Error',
    FranchaiseeDetectChanges = '[FranchaiseeDetectChanges] Changes',
    CloseNewFranchaiseeModal = '[CloseNewFranchaiseeModal] Close Franchaisee Modal',
    SearchFranchaisee = '[SearchFranchaisee] Search',
    SearchFranchaiseeSuccess = '[SearchFranchaiseeSuccess] Search Result'
}

export class CloseNewFranchaiseeModal implements Action {
  readonly type = FranchaiseeActionTypes.CloseNewFranchaiseeModal;
  constructor(public payload: boolean) {}
}

export class FranchaiseeDetectChanges implements Action {
  readonly type = FranchaiseeActionTypes.FranchaiseeDetectChanges;
  constructor(public payload: boolean) {}
}

export class CreateFranchaiseeSuccess implements Action {
  readonly type = FranchaiseeActionTypes.CreateFranchaiseeSuccess;
  constructor(public payload: boolean) {}
}

export class ShowNewFranchaiseeModal implements Action {
  readonly type = FranchaiseeActionTypes.ShowNewFranchaisee;
  constructor(public payload: boolean) {}
}

export class CreateFranchaisee implements Action {
  readonly type = FranchaiseeActionTypes.CreateFranchaisee;
  constructor(public payload: Franchaisee) {}
}

export class FranchaiseeError implements Action {
  readonly type = FranchaiseeActionTypes.FranchaiseeError;
  constructor(public payload: any) {}
}

export class SearchFranchaisee implements Action {
  readonly type = FranchaiseeActionTypes.SearchFranchaisee;
  constructor(public payload: string) {}
}

export class SearchFranchaiseeSuccess implements Action {
  readonly type = FranchaiseeActionTypes.SearchFranchaiseeSuccess;
  constructor(public payload: Franchaisee[]) {}
}

export type FranchaiseeAction = 
  CreateFranchaisee | 
  ShowNewFranchaiseeModal |
  CreateFranchaiseeSuccess |
  FranchaiseeDetectChanges |
  CloseNewFranchaiseeModal |
  FranchaiseeError |
  SearchFranchaisee |
  SearchFranchaiseeSuccess;
