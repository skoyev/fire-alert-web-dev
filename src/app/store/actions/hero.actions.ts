import { Action } from '@ngrx/store';

import { Hero } from '@appModels/hero';

export enum HeroActionTypes {
  heroGetHeroes = '[Hero] get',
  heroGetHeroesSuccess = '[Hero] get heroes success',
  heroAddHero = '[Hero] add hero',
  heroAddHeroSuccess = '[Hero] add hero success',
  heroDeleteHero = '[Hero] delete hero',
  heroDeleteHeroSuccess = '[Hero] delete hero success',
  heroGetHeroById = '[Hero] get hero by id',
  heroGetHeroByIdSuccess = '[Hero] get hero by id success',
  heroUpdateHero = '[Hero] update hero',
  heroUpdateHeroSuccess = '[Hero] update hero success',
  heroSearchHeroes = '[Hero] search heroes',
  heroSearchHeroesSuccess = '[Hero] search heroes success',
  heroSearchHeroesReset = '[Hero] search heroes reset',
  heroError = '[Hero] error'
}

export class GetHeroes implements Action {
  readonly type = HeroActionTypes.heroGetHeroes;
}

export class GetHeroesSuccess implements Action {
  readonly type = HeroActionTypes.heroGetHeroesSuccess;
  constructor(public payload: Hero[]) {}
}

export class AddHero implements Action {
  readonly type = HeroActionTypes.heroAddHero;
  constructor(public payload: Hero) {}
}

export class AddHeroSuccess implements Action {
  readonly type = HeroActionTypes.heroAddHeroSuccess;
  constructor(public payload: Hero) {}
}

export class GetHeroById implements Action {
  readonly type = HeroActionTypes.heroGetHeroById;
  constructor(public payload: number) {}
}

export class GetHeroByIdSuccess implements Action {
  readonly type = HeroActionTypes.heroGetHeroByIdSuccess;
  constructor(public payload: Hero) {}
}

export class UpdateHero implements Action {
  readonly type = HeroActionTypes.heroUpdateHero;
  constructor(public payload: Hero) {}
}

export class UpdateHeroSuccess implements Action {
  readonly type = HeroActionTypes.heroUpdateHeroSuccess;
  constructor(public payload: Hero) {}
}

export class DeleteHero implements Action {
  readonly type = HeroActionTypes.heroDeleteHero;
  constructor(public payload: Hero) {}
}

export class DeleteHeroSuccess implements Action {
  readonly type = HeroActionTypes.heroDeleteHeroSuccess;
  constructor(public payload: Hero) {}
}

export class SearchHeroes implements Action {
  readonly type = HeroActionTypes.heroSearchHeroes;
  constructor(public payload: string) {}
}

export class SearchHeroesSuccess implements Action {
  readonly type = HeroActionTypes.heroSearchHeroesSuccess;
  constructor(public payload: Hero[]) {}
}

export class SearchHeroesReset implements Action {
  readonly type = HeroActionTypes.heroSearchHeroesReset;
}

export class HeroError implements Action {
  readonly type = HeroActionTypes.heroError;
  constructor(public payload: any) {}
}

export type HeroActions =
  | GetHeroes
  | GetHeroesSuccess
  | AddHero
  | AddHeroSuccess
  | GetHeroById
  | GetHeroByIdSuccess
  | UpdateHero
  | UpdateHeroSuccess
  | DeleteHero
  | DeleteHeroSuccess
  | SearchHeroes
  | SearchHeroesSuccess
  | SearchHeroesReset
  | HeroError;
