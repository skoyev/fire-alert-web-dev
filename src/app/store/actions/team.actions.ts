import { Action } from '@ngrx/store';
import { Team } from '@appModels/team';

export enum TeamActionTypes {
  CreateTeam = '[TeamAction] Create'
}

export class CreateTeam implements Action {
  readonly type = TeamActionTypes.CreateTeam;
  constructor(public payload: Team) {}
}
export type TeamActions = CreateTeam;
