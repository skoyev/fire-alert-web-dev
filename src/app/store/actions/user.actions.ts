import { Action } from '@ngrx/store';

export enum UserActionTypes {
    ShowProfile = '[UserAction] ShowProfile'
}

export class ShowProfile implements Action {
  readonly type = UserActionTypes.ShowProfile;
  constructor(public payload: boolean) {}
}
export type UserActions = ShowProfile;
