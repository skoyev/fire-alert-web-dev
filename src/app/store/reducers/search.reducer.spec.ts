import * as fromReducer from './search.reducer';
import { Search, SearchReset } from '@appStore/actions/search.actions';

describe('search reducer', () => {
  const init = { type: '@@init' } as any;
  let initialState: fromReducer.State;

  beforeEach(() => {
    initialState = {
      search: 'search'
    };
  });

  describe('State Changes', () => {
    it('should have an initial state', () => {
      const state = fromReducer.reducer(initialState, init);

      expect(state).toEqual(initialState);
    });

    it('should update search', () => {
      const search = new Search('new search');
      const state = [init, search].reduce(fromReducer.reducer, initialState);

      expect(state).toMatchSnapshot();
    });

    it('should reset search', () => {
      const reset = new SearchReset();
      const state = [init, reset].reduce(fromReducer.reducer, initialState);

      expect(state).toMatchSnapshot();
    });
  });
});
