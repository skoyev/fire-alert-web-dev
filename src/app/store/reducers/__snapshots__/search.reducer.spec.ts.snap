// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer State Changes should reset search 1`] = `
Object {
  "search": "",
}
`;

exports[`search reducer State Changes should update search 1`] = `
Object {
  "search": "new search",
}
`;
